### PR TITLE
Improve example lightstep_alert config to include alerting rule

### PR DESCRIPTION
**Pull request by lambcode**
Not having the alerting rule in the example has caused some confusion. Especially given there is a alerting_rule resource that is old and shouldn't be used anymore. Also added notes on the resources that will be deprecated.

### CODE REVIEW

File: docs/resources/alert.md [hunk 1]
```diff
@@ -39,6 +39,11 @@ spans latency
 | point percentile(value, 50)
 EOT
   }
+
+  alerting_rule {
+      id = lightstep_pagerduty_destination.my_destination.id
+  }
```
Comment (mattcarmody) [inline docs/resources/alert.md:45]: would it be helpful to readers to include a frequency?

File: docs/resources/stream_condition.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ description: |-
 
 ---
 
+# NOTE: this resource will be deprecated in v2+. Use `lightstep_alert` instead.
```
Comment (mattcarmody) [inline docs/resources/stream_condition.md:9]: while in here marking things for future deprecation, is `metric_condition` another candidate to be deprecated?

**Review by mattcarmody — APPROVED** [review]
🎉 love helpful docs

File: docs/resources/stream_condition.md (hunk 2 above)
Comment (lambcode) [reply docs/resources/stream_condition.md:9]: `metric_condition` already has a deprecation note on it :)

**Pull request merged**